File: 2017.1/2017-04-20/BroadcastReceiver/app/src/main/java/br/ufpe/cin/if1001/broadcastreceiver/MainActivity.java
package br.ufpe.cin.if1001.broadcastreceiver;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends Activity {

    public static final String INTENT_ACTION = "br.ufpe.cin.if1001.broadcasts.exemplo";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button enviarBroadcast = (Button) findViewById(R.id.enviarBroadcast);
        enviarBroadcast.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                sendBroadcast(new Intent(INTENT_ACTION),android.Manifest.permission.VIBRATE);
            }
        });

        Button abrirActivity = (Button) findViewById(R.id.abrirActivity);
        abrirActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(),DynRecActivity.class));
            }
        });

        Button bateriaActivity = (Button) findViewById(R.id.bateriaActivity);
        bateriaActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getApplicationContext(),BateriaActivity.class));
            }
        });



    }
}

File: 2019-04-15/Services/app/src/main/java/br/ufpe/cin/android/services/MainActivity.kt
package br.ufpe.cin.android.services

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        botao_mainThread.setOnClickListener { startActivity(Intent(applicationContext, MainThreadServiceActivity::class.java)) }

        botao_musicPlayer.setOnClickListener { startActivity(Intent(applicationContext, MusicPlayerActivity::class.java)) }

        botao_musicServiceWithoutBinding.setOnClickListener { startActivity(Intent(applicationContext, MusicPlayerNoBindingActivity::class.java)) }

        botao_musicServiceWithBinding.setOnClickListener { startActivity(Intent(applicationContext, MusicPlayerWithBindingActivity::class.java)) }

        botao_downloadService.setOnClickListener { startActivity(Intent(applicationContext, DownloadActivity::class.java)) }

        botao_broadcast.setOnClickListener { sendBroadcast(Intent("br.ufpe.cin.android.broadcasts.exemplo")) }

    }
}
